app/builders/application_form_builder.rb

```diff
@@ -0,0 +1,78 @@
+class ApplicationFormBuilder < ActionView::Helpers::FormBuilder
+  include ActionView::Helpers::TagHelper
+
+  STYLES = {
+    default: "bg-transparent border border-2 border-white rounded-md text-white
+    placeholder-white/50 focus:bg-white focus:text-black focus:placeholder-gray
+    focus:border-white focus:ring-white transition-all",
+    toggle_field: "relative w-11 h-6 bg-gray-6 rounded-full peer dark:bg-gray-6
+    peer-checked:after:translate-x-full rtl:peer-checked:after:-translate-x-full
+    peer-checked:after:border-white after:content-[''] after:absolute after:top-[2px]
+    after:start-[2px] after:bg-white after:rounded-full after:h-5 after:w-5
+    after:transition-all peer-checked:bg-red cursor-pointer",
+    error_message: "text-red-500 text-sm mt-1",
+    label: "font-bold text-base"
+  }.freeze
+
+  ERROR_STYLES = {
+    default: "ring-1 ring-red-500 !focus:ring-red-500"
+  }.freeze
+
+  [
+    :text_field,
+    :check_box,
+    :date_field,
+    :email_field,
+    :number_field,
+    :search_field,
+    :password_field,
+    :text_area
+  ].each do |method_name|
+    define_method(method_name) do |attribute, options = {}|
+      options = add_base_styles(options, method_name, attribute)
+      super(attribute, options)
+    end
+  end
+
+  def label(attribute, content_or_options = nil, options = {}, &block)
+    options = add_base_styles(options, __method__, attribute)
+    super
+  end
+
+  def select(attribute, choices = nil, options = {}, html_options = {})
+    options = add_base_styles(options, __method__, attribute)
+    super
+  end
+
+  def error_message(attribute, options = {})
+    return unless attribute_has_error?(attribute)
+
+    options = add_base_styles(options, __method__, attribute)
+    error_message = @object.errors.full_messages_for(attribute).join(", ")
+    content_tag :div, error_message, options
+  end
+
+  def toggle_field(attribute, content_or_options = nil, options = {})
+    options = add_base_styles(options, __method__, attribute)
+    @template.render "shared/form_elements/toggle_field", form: self, attribute: attribute, options: options
+  end
+
+  private
+
+  def add_base_styles(options, method_name, attribute)
+    return options if options[:skip_base_styling].present?
+
+    style = []
+    style << options[:class]
+    style << (STYLES[method_name] || STYLES[:default])
+    style << (ERROR_STYLES[method_name] || ERROR_STYLES[:default]) if attribute_has_error?(attribute)
```

Comment:
I think that this line can be removed as the error styling is already being added by the classes in  "application.tailwind.css".

Reply:
I agree. The "application.tailwind.css" uses the default class `field_with_errors`. 

Note: We must be careful not to use some classes that may not be suitable for all inputs. If we would like to customize specific error styles per input field we may return to this option.